### PR TITLE
fix: support c10::List type of output in MarkOutputs

### PR DESCRIPTION
**Pull request by inocsin**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
Support c10::List<c10::IValue> type of output type in **void MarkOutputs(ConversionCtx* ctx, at::ArrayRef<const torch::jit::Value*> outputs)**

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: core/conversion/conversion.cpp [hunk 1]
```diff
@@ -173,24 +173,34 @@ void AddInputs(ConversionCtx* ctx, at::ArrayRef<const torch::jit::Value*> inputs
...
+    std::string name = std::string("output_") + std::to_string(ctx->num_outputs);
+    auto output_container = out_ivalue.toCustomClass<TensorContainer>();
+    nvinfer1::ITensor* out_tensor = output_container.get()->tensor();
+    out_tensor->setName(name.c_str());
+    ctx->net->markOutput(*out_tensor);
+    LOG_INFO(
+        ctx->logger, "Marking Output " << out->debugName() << " named " << name << " in engine (ctx.MarkOutput)");
+    ctx->num_outputs += 1;
+  } else {
+    TRTORCH_THROW_ERROR("Unknown output type. Only a single tensor or a TensorList type is supported.");
```
Comment (narendasan) [inline core/conversion/conversion.cpp:187]: We should make this Error a bit more clear, something like 

```
Unsupported output type, only Tensors or unwrapped collections of Tensors can be marked as engine outputs but found type: [Print type here]
```

File: core/conversion/conversion.cpp [hunk 2]
```diff
@@ -173,24 +173,34 @@ void AddInputs(ConversionCtx* ctx, at::ArrayRef<const torch::jit::Value*> inputs
   }
 #endif
 }
+void MarkOutputsOfIvalue(ConversionCtx* ctx, c10::IValue out_ivalue, const torch::jit::Value* out) {
```
Comment (narendasan) [inline core/conversion/conversion.cpp:176]: rename to `MarkIValueOutputs`

**Review by narendasan — COMMENTED** [review]
I am a bit concerned because say we have a TensorList Value `%out1`. The output of the TensorRT engine could have like 10 outputs associated with that one Value. If there is an additional Tensor Value output `%out2` the output of the engine will have 11 tensors but we don't really have a way to determine which are supposed to be in a list and which should be by themselves. Right now the operator interface for all engines is `trt::execute_engine(TRTEngine, Tensor[]) -> Tensor[]`. @peri044 I think we discussed  this regarding tuples as well. We would need to store more information about output formatting and change the graph construction step in `compile.cpp` to take the output of `trt::execute_engine` and reconstruct the expected output format, but that is a much larger PR i believe.

**Comment by peri044** [comment]
Currently this PR can be put on hold as it involves a broader change across other components like graph synthesis as described in https://github.com/NVIDIA/TRTorch/issues/428

**Pull request closed without merging**